lib/src/command/global_run.dart

```diff
@@ -57,8 +61,18 @@ class GlobalRunCommand extends PubCommand {
       log.warning('The --mode flag is deprecated and has no effect.');
     }
 
-    var exitCode = await globals.runExecutable(package, executable, args,
-        enableAsserts: argResults['enable-asserts'] || argResults['checked']);
+    final experiments = argResults['enable-experiment'] as List;
+    final vmArgs = <String>[
+      if (experiments.isNotEmpty) "--enable_experiment=${experiments.join(',')}"
+    ];
+    final globalEntrypoint = await globals.find(package);
+    final exitCode = await runExecutable(
+        globalEntrypoint, Executable(package, executable), args,
+        vmArgs: vmArgs,
+        enableAsserts: argResults['enable-asserts'] || argResults['checked'],
+        recompile: (executable) async {
+      await globalEntrypoint.precompileExecutable(executable);
```

Comment:
`recompile: globalEntrypoint.precompileExecutable` wouldn't work? I don't think you need to wrap it.